iron-list.html

```diff
@@ -1870,15 +1872,31 @@
         }
       }
     },
-
-    _didMoveUp: function() {
-      this._focusPhysicalItem(this._focusedIndex - 1);
+    // When `grid === true` the `downArrow` goes forward `this._itemsPerRow` items
+    _didMoveBackward: function(e) {;
+      this._focusPhysicalItem(this._focusedIndex - (!this.grid ? 1 : this._itemsPerRow));
     },
-
-    _didMoveDown: function(e) {
+    // When `grid === true` the `upArrow` goes backward `this._itemsPerRow` items
+    _didMoveForward: function(e) {;
+      var destination = this._focusedIndex + (!this.grid ? 1 : this._itemsPerRow);
+	  // When `grid === true` move focus down one row, either
+	  // directly below or the last item of the list.
+	  if (this.grid && this._focusedIndex < (this.items.length - (this.items.length % this._itemsPerRow))) {
+		  destination = Math.min(destination, this.items.length - 1);
+	  }
```

Comment:
This if statement can be eliminated if you set the correct destination in one statement:

```js
var destination = Math.min(this._focusedIndex + (this.grid ? this._itemsPerRow : 1), this._virtualCount - 1);
```

or


```js
var destination = this._focusedIndex + (this.grid ? this._itemsPerRow : 1);
destination = Math.min(destination, this._virtualCount - 1);
```